sentinel/src/androidTest/kotlin/com/infinum/sentinel/data/sources/raw/formatters/HtmlStringBuilderTests.kt

```diff
@@ -123,6 +121,33 @@ internal class HtmlStringBuilderTests {
         ?.use { it.readText() }
         .orEmpty()
 
+    private fun checkDeviceSpecificFields(html: String): String {
+        val fields = listOf(
+            "screen_width",
+            "screen_height",
+            "screen_size",
+            "screen_density",
+            "font_scale"
+        )
+
+        val fieldPatterns = fields.map { field ->
+            field to Regex("""<div>$field:\s*[^<]*</div>""")
+        }
+
+        var updatedHtml = html
+
+        fieldPatterns.forEach { (field, pattern) ->
+            val matchResult = pattern.find(updatedHtml)
+            if (matchResult == null) {
+                throw AssertionError("Field $field is missing in the device object")
+            } else {
+                updatedHtml = updatedHtml.replace(matchResult.value, """<div>$field: </div>""")
+            }
+        }
+
+        return updatedHtml
+    }
+
```

Comment:
When it comes to StringBuilder tests I didn't find an elegant way to mock these fields like it is the case with `ReflectionHelpers`. So I went with this approach because I think it is only important to assert that field is there.

When I get more familiar with Roboelectric, I will return to this. unless you have suggestions.